### PR TITLE
Update documentation for TypeScript Model Definition & Class Fields

### PR DESCRIPTION
**Pull request by ephys**
### Pull Request Checklist

- [n/a] Have you added new tests to prevent regressions?
- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [n/a] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

There is an incompatibility between Sequelize & adding public class fields for attributes.

This PR documents how this incompatibility & explains how to add typing without introducing actual class fields that will cause bugs.

It also detects uses of public class fields that shadow a model's attributes, and warns the user. Doing this:

image

Will log the following to the console:

image

Closes https://github.com/sequelize/sequelize/issues/10579
Closes https://github.com/sequelize/sequelize/issues/12603
Closes https://github.com/sequelize/sequelize/issues/11675
Closes https://github.com/sequelize/sequelize/issues/11032
Closes https://github.com/sequelize/sequelize/issues/11494

---

Note: this documentation update is a temporary fix as things are going to break again once decorators are stage 4. I'll open an rfc with possible solutions that resolve this caveat for good for V7.

### CODE REVIEW

File: docs/manual/core-concepts/model-basics.md [hunk 1]
```diff
@@ -77,6 +77,69 @@ console.log(User === sequelize.models.User); // true
...
+}, { sequelize });
+
+const user = new User({ id: 1 });
+user.id; // 1
+```
+
+In TypeScript, you can add typing information without adding an actual public class field by using the `declare` keyword:
+
+```typescript
+// Invalid
```
Comment (sdepold) [inline docs/manual/core-concepts/model-basics.md:126]: is it invalid or valid? :D

**Review by sdepold — APPROVED** [review]

**Comment by sdepold** [comment]
do we want to have that in v7 as well and eventually remove it?

File: docs/manual/core-concepts/model-basics.md (hunk 1 above)
Comment (ephys) [reply docs/manual/core-concepts/model-basics.md]: Good catch! Changed, it is valid :)

**Comment by ephys** [comment]
Yes let's add it to v7 too until we've resolved https://github.com/sequelize/meetings/issues/16 :)

**Pull request closed without merging**